File: sirius-core/src/main/java/org/starcoin/util/FileUtil.java
package org.starcoin.util;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Comparator;

public class FileUtil {

    public static void writeFile(String filename, byte[] data) {
        writeFile(new File(filename), data);
    }

    public static void writeFile(File file, byte[] data) {
        try {
            FileOutputStream out = new FileOutputStream(file);
            out.write(data);
            out.close();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public static byte[] readFile(String filename) {
        return readFile(new File(filename));
    }

    public static byte[] readFile(File file) {
        try {
            int len = (int) file.length();
            byte[] data = new byte[len];
            DataInputStream in = new DataInputStream(new FileInputStream(file));
            in.read(data);
            in.close();

            return data;
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public static void deleteDir(File dir) {
        try {
            Files.walk(dir.toPath())
                    .map(Path::toFile)
                    .sorted(Comparator.reverseOrder())
                    .forEach(File::delete);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: sirius-core/src/main/kotlin/org/starcoin/sirius/util/FileUtil.kt
package org.starcoin.sirius.util

import java.io.File
import java.nio.file.Files
import java.util.*

@Deprecated("Directly use File extension function")
object FileUtil {

    fun writeFile(filename: String, data: ByteArray) {
        writeFile(File(filename), data)
    }

    fun writeFile(file: File, data: ByteArray) {
        file.writeBytes(data)
    }

    fun readFile(filename: String): ByteArray {
        return readFile(File(filename))
    }

    fun readFile(file: File): ByteArray {
        return file.readBytes()
    }

    fun deleteDir(dir: File) {
        Files.walk(dir.toPath())
            .map { it.toFile() }
            .sorted(Comparator.reverseOrder())
            .forEach { it.delete() }
    }
}

fun File.readBytes(): ByteArray = this.inputStream().use {
    it.readBytes()
}

fun File.writeBytes(bytes: ByteArray) = this.outputStream().use { it.write(bytes) }